docs/DeveloperGuide.md

```diff
@@ -6,33 +6,183 @@
 
 ## Design & implementation
```

Comment:
Good effort on the drawing, would be great to consider using a professional tool for diagrams instead.